javaparser-core/src/main/java/com/github/javaparser/ast/expr/LongLiteralExpr.java

```diff
@@ -107,6 +118,34 @@ public long asLong() {
         return Long.parseLong(result);
     }
 
+    /**
+     * This function returns a representation of the literal value as a number. This will return a long, except for the
+     * case when the literal has the value <code>9223372036854775808L</code>. This special literal is only allowed in
+     * the expression <code>-9223372036854775808L</code> which represents <code>Long.MIN_VALUE</code>). However
+     * 9223372036854775808 (2^63) is out of range of long, which is -(2^63) to (2^63)-1 and thus a BigInteger must be
+     * returned.
+     * <p>
+     * Note, that this function will NOT return a negative number if the literal was specified in decimal, since *
+     * according to the language specification an expression such as <code>-1L</code> is represented by a unary *
```

Comment:
Is this section 15.15.4 or a different section?

Reply:
It is defined in 3.10.1 https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.10.1. I can directly refer to the chapter.